msticpy/sectools/ip_utils.py

```diff
@@ -0,0 +1,200 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+
+Contains a series of functions required to manipulate and enrich IP Address data
+to assist investigations.
+
+Designed to support any data source containing IP address entity.
+
+"""
+import ipaddress as ip
+from functools import lru_cache
+from typing import Tuple
+
+import pandas as pd
+from ipwhois import IPWhois
+
+from .._version import VERSION
+from ..nbtools.entityschema import GeoLocation, IpAddress
+from ..nbtools.utility import export
+from .geoip import GeoLiteLookup
+
+__version__ = VERSION
+__author__ = "Ashwin Patil"
+
+IPLOCATION = GeoLiteLookup()
+
+
+class Error(Exception):
```

Comment:
I think it's good to have custom exceptions - I regret not doing that from the beginning. It would be good to have maybe more descriptive names though if you are going to create custom exceptions. 
Also I don't see that you've used them anywhere?

Reply:
Think this might be a carry over from syslog_utils where I have the DataErrror custom exception.

Reply:
yup, carried over from syslog_utils. I will check and remove if these are unused in ip_utils.